File: Python/3_ex/fibonacci_recursive.py
# Undergraduate Student: Arturo Burgos
# Professor: João Rodrigo Andrade
# Federal University of Uberlândia - UFU, Fluid Mechanics Laboratory - MFLab, Block 5P, Uberlândia, MG, Brazil


# Third exercise: Fibonacci sequence - by a common loop

#####################
# Recursive fibonacci  --> Here I call the function itself in the same function
#####################

import time


n = int(input("Enter the n indices: "))



def recur_fibonacci(k):
    """
      Find the Fibonacci number of order n by recursion
    """
    if k < 2:
        return k
    else:
        return recur_fibonacci(k-1) + recur_fibonacci(k-2)

initial_time = time.time()
Fbn = recur_fibonacci(n)

print("The elapsed time is: %s seconds" % (time.time()-initial_time))
print("The correspond indices number is: ",Fbn)

File: Julia/3_ex/fibonacci_recursive.jl
# Undergraduate Student: Arturo Burgos
# Professor: João Rodrigo Andrade
# Federal University of Uberlândia - UFU, Fluid Mechanics Laboratory - MFLab, Block 5P, Uberlândia, MG, Brazil


# Third exercise: Fibonacci sequence - by a common loop

using Printf
using Statistics
using CPUTime

println("Choose a indice value for n: ")
N = chomp(readline())
n = parse(Int32,N)
println("\n")

Fbn = 0

##############################
# Variable iterative fibonacci  --> Worse performance, why?
##############################

function recurfibonacci(k)

    if k <= 2
     return 1
    else
       return S = recurfibonacci(k-1) + recurfibonacci(k-2)
    end
    
end

Fbn = recurfibonacci(n)

println("The correspond indices number is: ",Fbn)

@time @CPUtime recurfibonacci(n)
